pandas/io/stata.py

```diff
@@ -572,7 +572,13 @@ def _cast_to_stata_types(data: DataFrame) -> DataFrame:
     """
     ws = ""
     # original, if small, if large
-    conversion_data = (
+    conversion_data: tuple[
+        tuple[type, type, type],
+        tuple[type, type, type],
+        tuple[type, type, type],
+        tuple[type, type, type],
+        tuple[type, type, type],
```

Comment:
Great that this works! Since all elements are the same, maybe use this much shorter type `tuple[tuple[type, ...], ...]`.

Reply:
I was thinking about that but on the other hand https://peps.python.org/pep-0484/ states that:

`Arbitrary-length homogeneous tuples can be expressed using one type and ellipsis, for example Tuple[int, ...]`

This tuple does not have arbitrary length (we know already that it has 3 elements), so maybe it would be better to leave it like that? Of course can change if you think that it would be better this way

Reply:
It is definitely the more precise type :) Happy to leave it as-is. It just looks weird when the type definition is as long as the actual tuple.